src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/RuntimeMetrics.cs

```diff
@@ -188,17 +190,18 @@ private static IEnumerable<Measurement<long>> GetGarbageCollectionCounts()
             }
         }
 
-        // TODO - Uncomment once an implementation for https://github.com/dotnet/runtime/issues/104844 is available.
-        //private static IEnumerable<Measurement<double>> GetCpuTime()
-        //{
-        //    if (OperatingSystem.IsBrowser() || OperatingSystem.IsTvOS() || OperatingSystem.IsIOS())
-        //        yield break;
+        private static IEnumerable<Measurement<double>> GetCpuTime()
+        {
+            Debug.Assert(s_processCpuTime is not null);
+            Debug.Assert(!OperatingSystem.IsBrowser() && !OperatingSystem.IsTvOS() && !(OperatingSystem.IsIOS() && !OperatingSystem.IsMacCatalyst()));
 
-        //    ProcessCpuUsage processCpuUsage = Environment.CpuUsage;
+            #pragma warning disable CA1416 // This call site is reachable on all platforms. 'Environment.CpuUsage' is unsupported on: 'ios', 'tvos'
```

Comment:
If you put the unsupported attributes on this method, would that take care of it?

Reply:
I'll try that and look if I get any failures in the CI.

Reply:
Looks like this worked fine. Thanks for the suggestion. It is cleaner and explanatory.